engines/illusions/detection.cpp

```diff
@@ -0,0 +1,206 @@
+	{ "illusions", "Illusions engine game" },
+	{ "bbdou", "Beavis and Butt-head Do U" },
+	{ "duckman", "Duckman" },
+	{ 0, 0 }
+};
+
+namespace Illusions {
+
+static const IllusionsGameDescription gameDescriptions[] = {
+	{
+		{
+			"bbdou",
+			0,
+			AD_ENTRY1s("000D0001.scr", "d0c846d5dccc5607a482c7dcbdf06973", 601980),
+			Common::EN_ANY,
+			Common::kPlatformWindows,
+			ADGF_UNSTABLE | ADGF_DROPPLATFORM,
+			GUIO0()
+		},
+		kGameIdBBDOU
+	},
+
+	{
+		{
+			"duckman",
+			0,
+			AD_ENTRY1s("duckman.gam", "172c0514f3793041718159cf9cf9935f", 29560832),
+			Common::EN_ANY,
+			Common::kPlatformWindows,
+			ADGF_DROPPLATFORM,
+			GUIO0()
+		},
+		kGameIdDuckman
+	},
+
+	{
+		{
+			"duckman",
+			0,
+			AD_ENTRY1s("duckman.gam", "64d16922ffb46b746fc2c12a14d75bcc", 29779968),
+			Common::DE_DEU,
+			Common::kPlatformWindows,
+			ADGF_DROPPLATFORM,
+			GUIO0()
+		},
+		kGameIdDuckman
+	},
+
+	{AD_TABLE_END_MARKER, 0}
+};
+
+} // End of namespace Illusions
+
+static const char * const directoryGlobs[] = {
+	"resource",
+	0
+};
+
+class IllusionsMetaEngine : public AdvancedMetaEngine {
+public:
+	IllusionsMetaEngine() : AdvancedMetaEngine(Illusions::gameDescriptions, sizeof(Illusions::IllusionsGameDescription), illusionsGames) {
+		_singleId = "illusions";
+		_maxScanDepth = 2;
+		_directoryGlobs = directoryGlobs;
+	}
+
+	virtual const char *getName() const {
+		return "Illusions";
+	}
+
+	virtual const char *getOriginalCopyright() const {
+		return "(C) The Illusions Gaming Company";
+	}
+
+	virtual bool hasFeature(MetaEngineFeature f) const;
+	virtual bool createInstance(OSystem *syst, Engine **engine, const ADGameDescription *desc) const;
+	virtual int getMaximumSaveSlot() const;
+	virtual SaveStateList listSaves(const char *target) const;
+	SaveStateDescriptor querySaveMetaInfos(const char *target, int slot) const;
+	virtual void removeSaveState(const char *target, int slot) const;
+};
+
+bool IllusionsMetaEngine::hasFeature(MetaEngineFeature f) const {
+	return
+		(f == kSupportsListSaves) ||
+		(f == kSupportsDeleteSave) ||
+		(f == kSupportsLoadingDuringStartup) ||
+		(f == kSavesSupportMetaInfo) ||
+		(f == kSavesSupportThumbnail) ||
+		(f == kSavesSupportCreationDate);
+}
+
+void IllusionsMetaEngine::removeSaveState(const char *target, int slot) const {
+	Common::String fileName = Common::String::format("%s.%03d", target, slot);
+	g_system->getSavefileManager()->removeSavefile(fileName);
+}
+
+int IllusionsMetaEngine::getMaximumSaveSlot() const {
+	return 999;
+}
+
+SaveStateList IllusionsMetaEngine::listSaves(const char *target) const {
+	Common::SaveFileManager *saveFileMan = g_system->getSavefileManager();
+	Illusions::IllusionsEngine::SaveHeader header;
+	Common::String pattern = target;
+	pattern += ".???";
+	Common::StringArray filenames;
+	filenames = saveFileMan->listSavefiles(pattern.c_str());
+	Common::sort(filenames.begin(), filenames.end());	// Sort (hopefully ensuring we are sorted numerically..)
```

Comment:
This is outdated copypasta, see e.g. commit 8f6d8f7f68002d559a9c62c863edfc65c9dc8618.